x/gamm/pool-models/balancer/util_test.go

```diff
@@ -0,0 +1,42 @@
+package balancer_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store/rootmulti"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+func createTestPool(t *testing.T, poolAssets []balancer.PoolAsset, swapFee, exitFee string) types.PoolI {
+	swapFeeDec, err := sdk.NewDecFromStr(swapFee)
+	require.NoError(t, err)
+
+	exitFeeDec, err := sdk.NewDecFromStr(exitFee)
+	require.NoError(t, err)
+
+	pool, err := balancer.NewBalancerPool(1, balancer.PoolParams{
+		SwapFee: swapFeeDec,
+		ExitFee: exitFeeDec,
+	}, poolAssets, "", time.Now())
+
+	require.NoError(t, err)
+
+	return &pool
+}
+
+func createTestContext(t *testing.T) sdk.Context {
+	db := dbm.NewMemDB()
+	logger := log.NewNopLogger()
+
+	ms := rootmulti.NewStore(db, logger)
+
+	return sdk.NewContext(ms, tmtypes.Header{}, false, logger)
+}
```

Comment:
nit: we can also add these to app/app_testing/test_suite.go!

Reply:
The files in `test_suite.go` seem to belong to the keeper test suite.

On the other hand, these are low-level balancer pool test utilities. I briefly looked at the `test_uite.go`, and I don't think there are any applications for these utilities there. So I think we might want to keep them here. Let me know if you agree

Reply:
oh yeah agree! Forgot that we're not using test suite while I was writing that suggestion!